packages/dapp/src/reducers/ui.ts

```diff
@@ -0,0 +1,12 @@
+import { AnyAction } from "redux";
+import { uiActions } from "../actionCreators/ui";
+import { Map } from "immutable";
+
+export function ui(state: Map<string, any> = Map<string, any>(), action: AnyAction): Map<string, any> {
```

Comment:
I added action/reducer for storing UI state in Redux, which I think we'll need at some point.

I didn't go down the path of scoping out what the UI state structured data should look like, so right now it's a simple Map that holds key/vals that the multiple View containers/components would be interested in.

@walker Thoughts on this?

We could enhance this by coming up with some sort of structure for the UI state data... like maybe `ui` breaks down into something like:

```
ui.pages = Map<string, Map<string, any>]
// Top level map uses the Component referenced by `Router` as the key.
// The internal maps are key/val pairs relevant to the Component. We could additionally have a 'global' namespace for key/vals that are shared throughout the entire dapp

Or at the very least, keep it simple like in this PR but create an enum with the keys for UI state values.

Reply:
I think we'll definitely need a ui reducer in the dapp